File: Practice Arrays/23Cops.py
def rec(m,x,y,arr):
    res=[True]*101
    res[0]=False
    arr.sort()
    cover=x*y
    end=0  
    for i in arr:
        [curr_s,curr_e]=[i-cover if i-cover>=1 else 1,i+cover if i+cover<=100 else 100]
        start=max(curr_s,end)
        for i in range(start,curr_e+1):
            res[i]=False
        end=curr_e
    return sum(res)

# cook your dish here
t=int(input())
for _ in range(t):
    m,x,y=map(int,input().split())
    arr=list(map(int,input().split()))
    print(rec(m,x,y,arr))






        
    

File: Practice Arrays/codigo_cs/23Cops.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Test23
{

    static int Rec(int m, int x, int y, List<int> arr)
    {
        bool[] res = Enumerable.Repeat(true, 101).ToArray();
        res[0] = false;
        arr.Sort();
        int cover = x * y;
        int end = 0;

        foreach (int i in arr)
        {
            int curr_s = Math.Max(i - cover, 1);
            int curr_e = Math.Min(i + cover, 100);
            int start = Math.Max(curr_s, end + 1);

            for (int j = start; j <= curr_e; j++)
            {
                res[j] = false;
            }

            end = curr_e;
        }

        return res.Count(b => b);
    }
    public static void Cops()

    {
        // your code goes here


        int t = int.Parse(Console.ReadLine());
        for (int test = 0; test < t; test++)
        {
            int[] input = Console.ReadLine().Split().Select(int.Parse).ToArray();
            int m = input[0], x = input[1], y = input[2];
            List<int> arr = Console.ReadLine().Split().Select(int.Parse).ToList();

            Console.WriteLine(Rec(m, x, y, arr));
        }

    }
}
